MOV.ino

```diff
@@ -0,0 +1,1077 @@
+
+// s points to an array of bytes that should be displayed. len is the number of bytes in that array to display. 
+
+// shift will shift the display `shift` columns to the left. By shifting through we can smooth scroll it across the width of the char. 
+// when shift=0 that means start at col 0, which means send the full char width of the first char.
+
+// Returns true if there is more left in the buffer that did not fit on the display.
+
+ byte updateLEDs( const byte *s , unsigned int len , byte shift ) {
+ 
+  unsigned int pixel_count = PIXEL_COUNT;  // How many pixels left to fill on the display?
+
+  byte font_col =0;     // What column of the current font char are we currently on? 
+
+  while (len && pixel_count) {
+
+    if (shift) {
+
+      // Use up all the shift columns without actually sending them to the display
+      shift--;
+      
+    } else {
+
+      byte font_index = (*s) -ASCII_OFFSET;     // The letter in the font of the current byte to display
+      
+      sendCol( pgm_read_byte_near( &fontdata[font_index ][font_col] ) );    // The column in the font of the current letter (read from PROGMEM)
+      pixel_count--;
+
+    }
+
+    font_col++;
+
+    if (font_col==FONT_WIDTH) {       // Finished with this letter?
+
+      // Move on to next letter
+
+      font_col=0;       
+      
+      s++;
+      len--;
+
+    }
+
+  }
+       
+
+  // If the display is longer than the string, fill any remaining pixels with blanks
+
+  while (pixel_count--) {
+    sendCol(  0  );    // All pixels in column off               
+  }
+    
+  // Latch everything we just sent into the pixels so it is actually displayed
+  show();
+
+  return len>0;     // If len>0 then we ran out of pixels before we ran out of message to display
+}
+
+
+
+
+void setup() {
+
+  Serial.begin(9600);         // Set up the serial port for recieving
+
+  UCSR0B &= ~ _BV( TXEN0 );   // Disable the serial TX pin (digital pin 1) so we can use it as digital IO for an LED string
+                              // Note that you can remove this line if you want to have bidirectional serial communication, 
+                              // but then you will loose a string of LEDS and will need to make the font be only 6 pixels tall.
+                              
+  // initialize LED strip pins
+  
+  PIXEL_DDR |= 0xff;         // Set all 8 pins to output mode (note that digital pin 0 is controlled by the serial port so this has no effect on it)
+
+  // This blinking signal seems to reset some stuck pixels and leaves all outputs cleanly low
+  PIXEL_PORT |= 0xff;       // Set all outputs to 1
+  delay( 100);
+  PIXEL_PORT &= ~0xff;      // Set all outputs to 0
+  delay( 100);
+  PIXEL_PORT |= 0xff;       // Set all outputs to 1
+  delay( 100);
+  PIXEL_PORT &= ~0xff;      // Set all outputs to 0
+  delay( 100);
+
+
+  cli();
+
+ // Show something on startup so we know it is working
+ 
+  appendStringToBuffer( "  .....   booting up  .....    " );
```

Comment:
try this: make this "...looping...text" string and then create copies of buffer and buffer_len. So:

```
byte original_buffer[ BUFFER_SIZE ];
volatile unsigned int original_buffer_len = 0; 

void setup() {

....

appendStringToBuffer("...looping...text")

original_buffer = buffer
original_buffer_len = buffer_len

}

void loop() {

... 
(remove the appendStringToBuffer( "...looping...text" ); but everything else looks ok)
...

if (buffer_len == 0) {
buffer_len = original_buffer_len
buffer = original_buffer
}





}
```

I'm not familiar with C++ but I think that's the essence of what you want to do, instead of copying the string again and again and again, just make a copy of the buffer and revert back to the original when its done.